### PR TITLE
Alternative Privacy Consideration section

### PR DESCRIPTION
**Pull request by huitema**
@martinthomson This is kind of an alternative view of the Privacy considerations. It is somewhat more compact than PR #25.

### CODE REVIEW

File: draft-thomson-train-protocol.md
Comment (martinthomson) [inline draft-thomson-train-protocol.md]:
```suggestion
# Privacy Considerations {#privacy}
```

File: draft-thomson-train-protocol.md
Comment (martinthomson) [inline draft-thomson-train-protocol.md]:
```suggestion
The focus of this analysis is the extent to which observing TRAIN
packets could be used to gain information about endpoints.
This might be leaking details of how applications using QUIC
operate or leaks of endpoint identity when using additional
privacy protection, such as a VPN.
```

File: draft-thomson-train-protocol.md
Comment (martinthomson) [inline draft-thomson-train-protocol.md]:
```suggestion
the data rate limits set by TRAIN.

There are two avenues of attack that require more analysis:

* that the passive observation of TRAIN packets might help identify or
  distinguish endpoints; and
* that active manipulation of TRAIN signals might help reveal the
  identity of endpoints that are otherwise hidden behind VPNs or proxies.
```

File: draft-thomson-train-protocol.md
Comment (martinthomson) [inline draft-thomson-train-protocol.md]:
```suggestion
a few of these servers use TRAIN, any TRAIN packets
```

File: draft-thomson-train-protocol.md
Comment (martinthomson) [inline draft-thomson-train-protocol.md]:
```suggestion
## Active manipulation of TRAIN to identify endpoints
```

File: draft-thomson-train-protocol.md
Comment (martinthomson) [inline draft-thomson-train-protocol.md]:
```suggestion
Suppose a configuration in which multiple clients use a VPN or proxy
```

File: draft-thomson-train-protocol.md
Comment (martinthomson) [inline draft-thomson-train-protocol.md]:
```suggestion
Suppose now that the attacker selects a flow on the link between the
VPN/proxy and server. The attacker applies rate limit signals to TRAIN packets
in that flow. The attacker chooses a bandwidth that is
lower than the "natural" bandwidth of the connection. A reduction
in the rate of flows between client and VPN/proxy might allow
the attacker to link the altered flow to the client.

~~~ aasvg
+--------+
| Client |------.
+--------+       \      +-------+
                  '---->|       |            +--------+
+--------+              |  VPN  |<==========>|        |
| Client |------------->|   /   |<==========>| Server |
+--------+              | Proxy |<==========>|        |
                  .---->|       |     ^      +--------+
+--------+       /      +-------+     |
| Client |======'                     |
+--------+      ^           Apply rate limit signal
                 \
                  \
               Observe change
~~~
```

File: draft-thomson-train-protocol.md
Comment (martinthomson) [inline draft-thomson-train-protocol.md]: I don't think we should say anything about opinions.

```suggestion
An attacker that can manipulate TRAIN headers can also simulate
congestion signals by dropping packets or by setting the ECN CE bit.
That will also likely result in changes in the congestion response by
the affected client.
```

File: draft-thomson-train-protocol.md
Comment (martinthomson) [inline draft-thomson-train-protocol.md]:
```suggestion
A VPN or proxy could defend against this style of attack by removing TRAIN (and
ECN) signals. There are few reasons to provide per-flow rate limit signals in
that situation.  Endpoints might also either disable this feature or ignore any
signals when they are aware of the use of a VPN or proxy.
```

**Review by martinthomson — APPROVED** [review]
Editorial suggestions only.  I like this better than mine.

File: draft-thomson-train-protocol.md
Comment (martinthomson) [inline draft-thomson-train-protocol.md]:
```suggestion
## Passive Attacks
```

File: draft-thomson-train-protocol.md
Comment (martinthomson) [inline draft-thomson-train-protocol.md]:
```suggestion
## Active Attacks
```

**Review by martinthomson — COMMENTED** [review]
Oh, and some alternative (punchier) headings.

File: draft-thomson-train-protocol.md
Comment (huitema) [reply draft-thomson-train-protocol.md]: Github does not let me accept the suggestion. I will do it in a separate commit.

**Comment by huitema** [comment]
@martinthomson Thanks for the suggestions! I was a bit short on time, knew I should do a figure but skipped on it... I will be travelling tomorrow and out for the Labor Day weekend, so feel free to merge this. Or wait for @kazuho approval...

**Pull request merged**
